AndroidAnnotations/androidannotations-core/androidannotations/src/main/java/org/androidannotations/internal/core/handler/FragmentArgHandler.java

```diff
@@ -40,25 +45,43 @@
 import com.helger.jcodemodel.JFieldRef;
 import com.helger.jcodemodel.JFieldVar;
 import com.helger.jcodemodel.JMethod;
+import com.helger.jcodemodel.JOp;
 import com.helger.jcodemodel.JVar;
 
-public class FragmentArgHandler extends BaseAnnotationHandler<EFragmentHolder> {
+public class FragmentArgHandler extends BaseAnnotationHandler<EFragmentHolder>implements HasMethodInjection<EFragmentHolder> {
+
+	private final InjectHelper<EFragmentHolder> injectHelper;
 
 	public FragmentArgHandler(AndroidAnnotationsEnvironment environment) {
 		super(FragmentArg.class, environment);
+		injectHelper = new InjectHelper<>(validatorHelper, this, InjectHelper.ValidationMode.FRAGMENT);
 	}
 
 	@Override
 	public void validate(Element element, ElementValidation validation) {
-		validatorHelper.enclosingElementHasEFragment(element,  validation);
+		injectHelper.validate(FragmentArg.class, element, validation);
 
 		validatorHelper.isNotPrivate(element, validation);
 
-		validatorHelper.canBePutInABundle(element, validation);
+		Element param = element;
```

Comment:
I saw this construct multiple times, maybe it should be moved to a helper method?